From 80b8279981d57d848e8a844dedfd1f3e234e814f Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Thu, 28 Jan 2016 09:12:52 +0100
Subject: [PATCH 12/13] S Core: Fix issues for spi nor access

Add support for spi nor flash even if you have compiled the u-boot for
the boot from sd card. That is needed to access and write a boot loader
from the system to the spi nor flash.
There is although an issue fixed for the initialisation of the SD 2
interface.

Ticket #871
---
 board/freescale/mx6sxscore/mx6sxscore.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/board/freescale/mx6sxscore/mx6sxscore.c b/board/freescale/mx6sxscore/mx6sxscore.c
index 490955b..6dbc43d 100644
--- a/board/freescale/mx6sxscore/mx6sxscore.c
+++ b/board/freescale/mx6sxscore/mx6sxscore.c
@@ -241,7 +241,7 @@ int board_mmc_init(bd_t *bis)
         switch (i) {
             case 0:
                 imx_iomux_v3_setup_multiple_pads(
-                        usdhc3_pads, ARRAY_SIZE(usdhc2_pads));
+                        usdhc2_pads, ARRAY_SIZE(usdhc2_pads));
                 usdhc_cfg[0].sdhc_clk = mxc_get_clock(MXC_ESDHC2_CLK);
                 break;
             case 1:
@@ -296,7 +296,6 @@ void board_late_mmc_init(void)
 }
 #endif //CONFIG_FSL_ESDHC
 
-#ifdef CONFIG_SYS_USE_SPINOR
 iomux_v3_cfg_t const ecspi3_pads[] = {
 	MX6SX_PAD_SD4_DATA6__ECSPI3_MISO | MUX_PAD_CTRL(SPI_PAD_CTRL),
 	MX6SX_PAD_SD4_DATA5__ECSPI3_MOSI | MUX_PAD_CTRL(SPI_PAD_CTRL),
@@ -310,7 +309,6 @@ void setup_spinor(void)
 					 ARRAY_SIZE(ecspi3_pads));
 	gpio_direction_output(IMX_GPIO_NR(6, 12), 0);
 }
-#endif // CONFIG_SYS_USE_SPINOR
 
 static iomux_v3_cfg_t const gpio_pads[] = {
     MX6SX_PAD_LCD1_DATA18__GPIO3_IO_19 | MUX_PAD_CTRL(NO_PAD_CTRL),
@@ -551,8 +549,8 @@ int board_init(void)
 	setup_i2c(0, CONFIG_SYS_I2C_SPEED, 0x7f, &i2c_pad_info1);
 #endif
 
-#ifdef CONFIG_SYS_USE_SPINOR
 	setup_spinor();
+#ifdef CONFIG_SYS_USE_SPINOR
     printf("Hello World from SPINOR\n");
 #else
     printf("Hello World from SD Card\n");
-- 
1.9.1

